staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/managedfieldsupdater.go

```diff
@@ -80,7 +81,8 @@ func (f *managedFieldsUpdater) Apply(liveObj, appliedObj runtime.Object, managed
 		managed.Times()[fieldManager] = &metav1.Time{Time: time.Now().UTC()}
 	}
 	if object == nil {
-		object = liveObj
+		object = liveObj.DeepCopyObject()
+		internal.RemoveObjectManagedFields(object)
```

Comment:
This line is exercised here: https://github.com/kubernetes/kubernetes/pull/106195/files#diff-16ff3ca9b9a959368bf1ed86bbe146fda7df824426129eceea577a4b4b032bc6R1374

I just manually verified that it is hit. I'm wondering if you know of a better way to make the connection between the test and that line's coverage explicit? Currently the test only checks that `liveObj` is unchanged.

Reply:
You maybe need to write a more target unit-test, probably in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/managedfieldsupdater_test.go

Reply:
we want a test that not only hits this line, but if possible, would fail without the deepcopy and dropping managed fields

Reply:
Yeah, but this line is basically completely pointless because `FieldManager` always replaces it anyway

Reply:
If we want to test that it's being reset, we need to test without the `FieldManager`.

Reply:
I've pushed my attempt to address the above feedback in 99fd43965a37a3552acf67554790d386a4d7afbd please take a look. This commit has a test which adds a node at the end of the manager pipeline, but before `FieldManager`, so that we can directly test if the `liveObj` has been copied, and its managedFields removed.